src/rules/groupedImportsRule.ts

```diff
@@ -0,0 +1,171 @@
+        if (this.previousImportStatement != null) {
+            if (importStatement.type === this.previousImportStatement.type) {
+                if (importStatement.lineStart !== this.previousImportStatement.lineEnd + 1) {
+                    return true;
+                }
+            } else {
+                if (importStatement.type < this.previousImportStatement.type) {
+                    return true;
+                } else if (importStatement.lineStart !== this.previousImportStatement.lineEnd + 2) {
+                    return true;
+                }
+            }
+        }
+        this.previousImportStatement = importStatement;
+        return false;
+    }
+
+    private toImportStatement = (statement: ts.ImportDeclaration): ImportStatement => {
+        return {
+            lineEnd: this.sourceFile.getLineAndCharacterOfPosition(statement.getEnd()).line,
+            lineStart: this.sourceFile.getLineAndCharacterOfPosition(statement.getStart()).line,
+            statement,
+            type: this.getImportStatementType(statement),
+        };
+    }
+
+    private getImportStatementType(statement: ts.ImportDeclaration): ImportStatementType {
+        const path = this.getImportPath(statement);
+        if (path.charAt(0) === ".") {
+            if (path.charAt(1) === ".") {
+                return ImportStatementType.PARENT_DIRECTORY_IMPORT;
+            } else {
+                return ImportStatementType.CURRENT_DIRECTORY_IMPORT;
+            }
+        } else {
+            return ImportStatementType.LIBRARY_IMPORT;
+        }
+    }
+
+    private getImportPath(statement: ts.ImportDeclaration): string {
+        if (isTextualLiteral(statement.moduleSpecifier)) {
+            return statement.moduleSpecifier.text;
+        }
+        return "";
+    }
+
+    private createFix(importStatements: ImportStatement[]): Lint.Fix {
+        const newLine = this.getEofChar(this.sourceFile);
+        const imports = this.getOrderedImports(importStatements);
+        const addition = Lint.Replacement.appendText(0, imports.join(newLine));
+        const deletions = importStatements.map((imp) => {
+            const [start, end] = this.getRangeIncludingWhitespace(imp.statement);
+            return Lint.Replacement.deleteFromTo(start, end);
+        });
+        return [...deletions, addition];
+    }
+
+    private getOrderedImports(importStatements: ImportStatement[]): string[] {
+        const libs = importStatements.filter((imp) => imp.type === ImportStatementType.LIBRARY_IMPORT);
+        const parent = importStatements.filter((imp) => imp.type === ImportStatementType.PARENT_DIRECTORY_IMPORT);
+        const current = importStatements.filter((imp) => imp.type === ImportStatementType.CURRENT_DIRECTORY_IMPORT);
+        return [libs, parent, current].reduce((arr, imps) => {
+            if (imps.length == 0) {
+                return arr;
+            }
+            return arr.concat(imps.map((imp) => imp.statement.getText()), "");
```

Comment:
nit: consider passing `sourceFile` as parameter to `.getText()` and `.getStart()` for performance reasons.
Though that's not as relevant here, because it's only used in case of a lint error (which should not be the hot path).